File: Task2/task2.swift
import Foundation
if let inputNumbers = readLine(),let numbers = Int(inputNumbers) {
    var clumba: [Int] = []
    var maxFlower = 0
    if numbers <= 0 {
        print("Быть не может")
        exit(0)
    }
    for _ in 0..<numbers {
        if let inputF = readLine(),let flower = Int(inputF) {
            if flower > 0 {
            clumba.append(flower)
        }
            maxFlower = max(maxFlower, flower)
        }else {
        print("Ошибка: Некорректный ввод.")
}
    }
    if let input = readLine(), let poliv = Int(input) {
        if poliv < maxFlower {
        print(" Как купишь новую лейку - приходи.")
        exit(0)
        } 
        else {
            var step = 0
            var perem = poliv
            for (i, flower) in clumba.enumerated() {
                if perem < flower {
                    step += (i * 2 + 1)
                    perem = poliv
                } else {
                    step += 1
                }
                perem -= flower
            }
            print(step)
        }
    } else {
        print("Ошибка: Некорректный ввод.")
    }
} else {
    print("Ошибка: Некорректный ввод.")
}

File: Task2/task2.py
import sys
numbers = int(input())
max_flower = 0
if numbers <= 0:
    print("Быть не может")
    sys.exit()
else:
    clumba = []
    while(numbers>0):
        flower = int(input())
        if flower > 0:
            clumba.append(flower)
        if flower > max_flower:
            max_flower = flower
        numbers-=1
    poliv = int(input())
    if poliv < max_flower:
        print(" Как купишь новую лейку - приходи.")
        sys.exit() 
    else:
        step = 0
        perem = poliv
        for i in range(0, len(clumba)):
            if perem < clumba[i]:
                step = step + (i * 2 + 1)
                perem = poliv
            else:
                step += 1
            perem -= clumba[i]
        print(step)
